overlord/configstate/configcore/refresh.go

```diff
@@ -38,6 +47,7 @@ func validateRefreshSchedule(tr Conf) error {
 		if _, err = timeutil.ParseSchedule(refreshTimerStr); err != nil {
 			return err
 		}
+		scheduleSet = true
```

Comment:
It's a bit strange to do this here, in a validation function, and the places where this is triggered also don't look obvious at a glance. I suggest a more general solution to this problem: let's do the EnsureBefore as the first line of the Run function inside this same package, in a defer so that it does that everytime, and for whatever configuration tweaks any of the functions may have done.